sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/parallel_component.py

```diff
@@ -227,7 +235,13 @@ def _attr_type_map(cls) -> dict:
 
     def _to_dict(self) -> Dict:
         """Dump the parallel component content into a dictionary."""
-        return convert_ordered_dict_to_dict({**self._other_parameter, **super(ParallelComponent, self)._to_dict()})
+        tmp_dict = convert_ordered_dict_to_dict({**self._other_parameter, **super(ParallelComponent, self)._to_dict()})
+        # The backend treats the partition keys as a serialized json string.
```

Comment:
@bupt-wenxiaole please move this to _to_rest_object instead of _to_dict.

to_dict should be aligned with schema

Reply:
@elliotzh  I also added the converter in sdk\ml\azure-ai-ml\azure\ai\ml\entities\_builders\parallel.py _to_rest_object()

Reply:
fixed